libs/x-content/src/test/java/org/elasticsearch/common/xcontent/MapXContentParserTests.java

```diff
@@ -91,7 +91,7 @@ public void compareTokens(CheckedConsumer<XContentBuilder, IOException> consumer
             try (XContentParser parser = createParser(xContentType.xContent(), BytesReference.bytes(builder))) {
                 try (XContentParser mapParser = new MapXContentParser(
                     xContentRegistry(), LoggingDeprecationHandler.INSTANCE, map, xContentType)) {
-                    assertEquals(parser.contentType(), mapParser.contentType());
+//                    assertEquals(parser.contentType(), mapParser.contentType());
```

Comment:
mapParser keeps the original provided contentType - for instance VND_JSON
parser.contentType in this test is created from `createParser(xContentType.xContent()` which looses the information about the original VND_JSON